packages/wonder-blocks-data/src/util/get-gql-request-id.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+import type {GqlOperation, GqlContext} from "./gql-types.js";
+
+const toString = (valid: mixed): string => {
+    if (typeof valid === "string") {
+        return valid;
+    }
+    return JSON.stringify(valid) ?? "";
+};
+
+/**
+ * Get an identifier for a given request.
+ */
+export const getGqlRequestId = <TData, TVariables: {...}>(
+    operation: GqlOperation<TData, TVariables>,
+    variables: ?TVariables,
+    context: GqlContext,
+): string => {
+    // We add all the bits for this into an array and then join them with
+    // a chosen separator.
+    const parts = [];
+
+    // First, we push the context values.
+    const sortedContext = Object.keys(context)
+        .sort()
+        .map((key) => `${key}=${context[key]}`)
+        .join(";");
+    parts.push(sortedContext);
```

Comment:
Differs from webapp since:

1. We don't know what the context may have (in webapp, we know what keys we have)
2. Includes the key name in the output; not just the context value